### ISSUE TITLE
nextflow log files not updating while wtp running

### ISSUE THREAD

**Issue by l-r-den** [replikation/What_the_Phage#161]
Hi,

Does the nextflow process execution timeline continue to update through the entire wtp run (i.e. should all finished processes be showing up in the timeline as they're completed)?

Along those lines, should the .nextflow.log files be updating after 'nextflow.script.ScriptRunner - > Launching execution'?

wtp has been running overnight with processes still in progress, but since the log files stopped updating I was wondering if it's a sign of something not executing properly or if this is the norm.

Thanks in advance

**Comment by mult1fractal** [comment]
Hey,

I never had such a long run O.O 
via htop you could check your pc workload  and see if WtP is doing some heavy computing.

You could try to stop the run (ctrl+c)  and then use your nextflow run command and add the -resume flag and then check if the log files are updating again.

the process execution timeline(and report).html is created after a finished run and is not being updated during a run.

I think you could also split your dataset into subsets to create smaller runs.

**Comment by l-r-den** [comment]
Ok, I tried -resume and it might be working now (prodigal command is running). Also if it matters, I misread and I'm using closer to 20 cores

I'll try splitting the assembly file. With multiple input files does wtp run them one at a time?

Thanks again for the input

**Comment by replikation** [comment]
@l-r-den depending on the system WtP tryies to parallel run samples and analyses. but it depends on the process (some need more cpu/ram). you can control this with the max_cores and cores flag.

e.g. on a 16 thread system (setting max_cores to 16) and using --cores 8 means it will run at most 2 processes in parallel. on a cluster/cloud there is no such limitation

**Comment by mult1fractal** [comment]
Hey
In the previous iteration, we used the most likely phage sequences from the Phage prediction  process for the annotation part.

But there is the possibility the prediction tools could miss a contig... so it is more reasonable to annotate the whole input fasta file. This way all sequences will be annotated and possible phage genes can also be annotated on contigs that were not predicted by the phage tools so the user can further analyze them.

**TLDR;**
You can use your input fastas for the --annotate process.

mhhh ok that's weird(markdown report fail)... let me test that again (--annotate only) but when I released v1.1.0 it worked

**Comment by mult1fractal** [comment]
Ok I tested the --annotate part and it works
`nextflow run replikation/What_the_Phage --fasta test-data/T7_draft.fa --annotate --output results/git_issue_annotate --databases /media/6tb_1/Databases/nextflow-autodownload-databases/ --cores 20 -profile local,docker -work-dir /media/6tb_1/work/ -r v1.1.0`

Idk... maybe the Vibrant fasta file is broken?

**Comment by l-r-den** [comment]
Ok thanks for testing that out. I'll try --annotate on some other fasta file and see what happens.



> Hey In the previous iteration, we used the most likely phage sequences from the Phage prediction process for the annotation part.
> 
> But there is the possibility the prediction tools could miss a contig... so it is more reasonable to annotate the whole input fasta file. This way all sequences will be annotated and possible phage genes can also be annotated on contigs that were not predicted by the phage tools so the user can further analyze them.
> 
> **TLDR;** You can use your input fastas for the --annotate process.
> 
> mhhh ok that's weird(markdown report fail)... let me test that again (--annotate only) but when I released v1.1.0 it worked

So wtp --annotate can take either full contigs with multiple phage genes on them and/or individual phage genes from the same contig?

 I didn't see a vibrant output file of the full input contigs that it found phage genes on & was thinking of pulling those out of my assembly and trying --annotate again with those
